File: backend/apps/products/management/commands/demo_data.py
"""
Management command to seed the database with demo product data.

Usage:
    python manage.py seed_products

This will create sellers and products matching the frontend demo data.
"""

from django.core.management.base import BaseCommand
from django.utils import timezone
from apps.users.models import User, SellerProfile, BuyerProfile
from apps.products.models import Product, Review
from decimal import Decimal


class Command(BaseCommand):
    help = 'Seed database with demo products and sellers'

    def add_arguments(self, parser):
        parser.add_argument(
            '--clear',
            action='store_true',
            help='Clear existing products and sellers before seeding',
        )

    def handle(self, *args, **options):
        self.stdout.write(self.style.SUCCESS('Starting product seeding...'))

        if options['clear']:
            self.stdout.write(self.style.WARNING('Clearing existing products and reviews...'))
            Review.objects.all().delete()
            Product.objects.all().delete()
            # Optionally clear sellers, but be careful
            # SellerProfile.objects.all().delete()
            # User.objects.filter(role=User.RoleChoices.SELLER).delete()

        # Demo sellers data
        sellers_data = [
            {
                'email': 'greenvalley@farm.com',
                'first_name': 'Green',
                'last_name': 'Valley',
                'store_name': 'Green Valley Farm',
                'pickup_address': '123 Farm Road, Farm City, Bangladesh',
                'phone_number': '+8801234567890',
                'nid_number': '1234567890123',
                'is_admin_verified': True,
                'is_email_verified': True,
                'is_profile_completed': True,
                'is_active': True,
            },
            {
                'email': 'sunrise@farm.com',
                'first_name': 'Sunrise',
                'last_name': 'Agriculture',
                'store_name': 'Sunrise Agriculture',
                'pickup_address': '456 Agriculture Lane, Farm District, Bangladesh',
                'phone_number': '+8801234567891',
                'nid_number': '1234567890124',
                'is_admin_verified': True,
                'is_email_verified': True,
                'is_profile_completed': True,
                'is_active': True,
            },
            {
                'email': 'naturesbest@farm.com',
                'first_name': 'Nature\'s',
                'last_name': 'Best',
                'store_name': 'Nature\'s Best Farm',
                'pickup_address': '789 Organic Street, Green Valley, Bangladesh',
                'phone_number': '+8801234567892',
                'nid_number': '1234567890125',
                'is_admin_verified': True,
                'is_email_verified': True,
                'is_profile_completed': True,
                'is_active': True,
            },
            {
                'email': 'tropical@farm.com',
                'first_name': 'Tropical',
                'last_name': 'Farm',
                'store_name': 'Tropical Farm',
                'pickup_address': '321 Fruit Garden Road, Coastal Area, Bangladesh',
                'phone_number': '+8801234567893',
                'nid_number': '1234567890126',
                'is_admin_verified': True,
                'is_email_verified': True,
                'is_profile_completed': True,
                'is_active': True,
            },
        ]

        # Create or get sellers
        sellers = []
        for seller_data in sellers_data:
            user, created = User.objects.get_or_create(
                email=seller_data['email'],
                defaults={
                    'first_name': seller_data['first_name'],
                    'last_name': seller_data['last_name'],
                    'phone_number': seller_data['phone_number'],
                    'role': User.RoleChoices.SELLER,
                    'is_admin_verified': seller_data['is_admin_verified'],
                    'is_email_verified': seller_data['is_email_verified'],
                    'is_profile_completed': seller_data['is_profile_completed'],
                    'is_active': seller_data['is_active'],
                }
            )
            
            if not created:
                # Update existing user
                for key, value in seller_data.items():
                    if key not in ['store_name', 'pickup_address', 'nid_number']:
                        setattr(user, key, value)
                user.save()

            # Create or update seller profile
            profile, profile_created = SellerProfile.objects.get_or_create(
                user=user,
                defaults={
                    'store_name': seller_data['store_name'],
                    'pickup_address': seller_data['pickup_address'],
                    'nid_number': seller_data['nid_number'],
                }
            )
            
            if not profile_created:
                profile.store_name = seller_data['store_name']
                profile.pickup_address = seller_data['pickup_address']
                profile.nid_number = seller_data['nid_number']
                profile.save()

            sellers.append(user)
            self.stdout.write(
                self.style.SUCCESS(f'✓ Seller: {seller_data["store_name"]} ({user.email})')
            )

        # Demo products data (matching frontend demoProducts)
        products_data = [
            {
                'name': 'Fresh Organic Tomatoes',
                'description': 'Premium organic tomatoes harvested fresh from local farms. Perfect for salads, cooking, and fresh consumption.',
                'price': Decimal('80.00'),
                'stock': 150,
                'unit': 'kg',
                'category': 'Vegetables',
                'image': 'https://images.unsplash.com/photo-1592841200221-a6898f307baa?w=800&h=600&fit=crop',
                'verified': True,
                'seller': sellers[0],  # Green Valley Farm
            },
            {
                'name': 'Farm Fresh Potatoes',
                'description': 'High quality potatoes, perfect for everyday cooking and various dishes.',
                'price': Decimal('45.00'),
                'stock': 200,
                'unit': 'kg',
                'category': 'Vegetables',
                'image': 'https://images.unsplash.com/photo-1518977822534-7049a61ee0c2?w=800&h=600&fit=crop',
                'verified': True,
                'seller': sellers[1],  # Sunrise Agriculture
            },
            {
                'name': 'Organic Green Beans',
                'description': 'Fresh, crisp green beans picked at peak freshness. Rich in nutrients.',
                'price': Decimal('120.00'),
                'stock': 80,
                'unit': 'kg',
                'category': 'Vegetables',
                'image': 'https://images.unsplash.com/photo-1459411621453-7b03977f4bfc?w=800&h=600&fit=crop',
                'verified': True,
                'seller': sellers[2],  # Nature's Best Farm
            },
            {
                'name': 'Fresh Mangoes',
                'description': 'Sweet, juicy mangoes from local orchards. Seasonal and delicious.',
                'price': Decimal('150.00'),
                'stock': 50,
                'unit': 'kg',
                'category': 'Fruits',
                'image': 'https://images.unsplash.com/photo-1605027990121-3fdb7a2e5b0f?w=800&h=600&fit=crop',
                'verified': True,
                'seller': sellers[3],  # Tropical Farm
            },
            {
                'name': 'Organic Red Onions',
                'description': 'Fresh red onions with strong flavor, perfect for cooking.',
                'price': Decimal('65.00'),
                'stock': 180,
                'unit': 'kg',
                'category': 'Vegetables',
                'image': 'https://images.unsplash.com/photo-1618512496242-a07f41e9c88e?w=800&h=600&fit=crop',
                'verified': True,
                'seller': sellers[0],  # Green Valley Farm
            },
            {
                'name': 'Fresh Cauliflower',
                'description': 'Premium quality cauliflower, white and fresh.',
                'price': Decimal('90.00'),
                'stock': 60,
                'unit': 'kg',
                'category': 'Vegetables',
                'image': 'https://images.unsplash.com/photo-1584270354949-b26bddf3342c?w=800&h=600&fit=crop',
                'verified': True,
                'seller': sellers[1],  # Sunrise Agriculture
            },
        ]

        # Create products
        created_count = 0
        updated_count = 0
        for product_data in products_data:
            product, created = Product.objects.get_or_create(
                name=product_data['name'],
                seller=product_data['seller'],
                defaults=product_data
            )
            
            if not created:
                # Update existing product
                for key, value in product_data.items():
                    setattr(product, key, value)
                product.save()
                updated_count += 1
            else:
                created_count += 1
            
            self.stdout.write(
                self.style.SUCCESS(f'✓ Product: {product_data["name"]} ({product_data["seller"].seller_profile.store_name})')
            )

        # Create some demo reviews for products
        # First, create a demo buyer
        buyer_email = 'freshmarket@example.com'
        buyer, buyer_created = User.objects.get_or_create(
            email=buyer_email,
            defaults={
                'first_name': 'Fresh',
                'last_name': 'Market',
                'role': User.RoleChoices.BUYER,
                'is_email_verified': True,
                'is_profile_completed': True,
                'is_active': True,
            }
        )
        
        if buyer_created:
            BuyerProfile.objects.create(
                user=buyer,
                business_name='Fresh Market Co.',
                delivery_address='456 Main Street, Dhaka, Bangladesh',
                nid_number='9876543210123',
            )

        # Create reviews
        reviews_data = [
            {'product_name': 'Fresh Organic Tomatoes', 'rating': 5, 'comment': 'Excellent quality tomatoes! Very fresh and delivered on time.'},
            {'product_name': 'Farm Fresh Potatoes', 'rating': 4, 'comment': 'Good quality potatoes. Would order again.'},
        ]

        for review_data in reviews_data:
            try:
                product = Product.objects.get(name=review_data['product_name'])
                Review.objects.get_or_create(
                    product=product,
                    buyer=buyer,
                    defaults={
                        'rating': review_data['rating'],
                        'comment': review_data['comment'],
                    }
                )
                self.stdout.write(
                    self.style.SUCCESS(f'✓ Review: {review_data["product_name"]} ({review_data["rating"]} stars)')
                )
            except Product.DoesNotExist:
                self.stdout.write(
                    self.style.WARNING(f'⚠ Product not found: {review_data["product_name"]}')
                )

        self.stdout.write(self.style.SUCCESS('\n' + '='*50))
        self.stdout.write(self.style.SUCCESS(f'✓ Created {created_count} new products'))
        self.stdout.write(self.style.SUCCESS(f'✓ Updated {updated_count} existing products'))
        self.stdout.write(self.style.SUCCESS(f'✓ Total sellers: {len(sellers)}'))
        self.stdout.write(self.style.SUCCESS('Product seeding completed successfully!'))
        self.stdout.write(self.style.SUCCESS('='*50))

File: frontend/src/data/demoData.js
// Demo data for all purposes when backend is not available

export const demoProducts = [
  {
    id: 1,
    name: 'Fresh Organic Tomatoes',
    description: 'Premium organic tomatoes harvested fresh from local farms. Perfect for salads, cooking, and fresh consumption.',
    price: 80,
    stock: 150,
    unit: 'kg',
    category: 'Vegetables',
    farmer_name: 'Green Valley Farm',
    seller_id: 1,
    farmer_id: 1,
    image: 'https://images.unsplash.com/photo-1592841200221-a6898f307baa?w=800&h=600&fit=crop',
    verified: true,
    rating: 4.8,
    reviews: 45,
  },
  {
    id: 2,
    name: 'Farm Fresh Potatoes',
    description: 'High quality potatoes, perfect for everyday cooking and various dishes.',
    price: 45,
    stock: 200,
    unit: 'kg',
    category: 'Vegetables',
    farmer_name: 'Sunrise Agriculture',
    seller_id: 2,
    farmer_id: 2,
    image: 'https://images.unsplash.com/photo-1518977822534-7049a61ee0c2?w=800&h=600&fit=crop',
    verified: true,
    rating: 4.6,
    reviews: 32,
  },
  {
    id: 3,
    name: 'Organic Green Beans',
    description: 'Fresh, crisp green beans picked at peak freshness. Rich in nutrients.',
    price: 120,
    stock: 80,
    unit: 'kg',
    category: 'Vegetables',
    farmer_name: 'Nature\'s Best Farm',
    seller_id: 3,
    farmer_id: 3,
    image: 'https://images.unsplash.com/photo-1459411621453-7b03977f4bfc?w=800&h=600&fit=crop',
    verified: true,
    rating: 4.9,
    reviews: 67,
  },
  {
    id: 4,
    name: 'Fresh Mangoes',
    description: 'Sweet, juicy mangoes from local orchards. Seasonal and delicious.',
    price: 150,
    stock: 50,
    unit: 'kg',
    category: 'Fruits',
    farmer_name: 'Tropical Farm',
    seller_id: 4,
    farmer_id: 4,
    image: 'https://images.unsplash.com/photo-1605027990121-3fdb7a2e5b0f?w=800&h=600&fit=crop',
    verified: true,
    rating: 4.7,
    reviews: 89,
  },
  {
    id: 5,
    name: 'Organic Red Onions',
    description: 'Fresh red onions with strong flavor, perfect for cooking.',
    price: 65,
    stock: 180,
    unit: 'kg',
    category: 'Vegetables',
    farmer_name: 'Green Valley Farm',
    seller_id: 1,
    farmer_id: 1,
    image: 'https://images.unsplash.com/photo-1618512496242-a07f41e9c88e?w=800&h=600&fit=crop',
    verified: true,
    rating: 4.5,
    reviews: 28,
  },
  {
    id: 6,
    name: 'Fresh Cauliflower',
    description: 'Premium quality cauliflower, white and fresh.',
    price: 90,
    stock: 60,
    unit: 'kg',
    category: 'Vegetables',
    farmer_name: 'Sunrise Agriculture',
    seller_id: 2,
    farmer_id: 2,
    image: 'https://images.unsplash.com/photo-1584270354949-b26bddf3342c?w=800&h=600&fit=crop',
    verified: true,
    rating: 4.6,
    reviews: 41,
  },
];

export const demoOrders = [
  {
    id: 1,
    buyer_name: 'Fresh Market Co.',
    product_name: 'Fresh Organic Tomatoes',
    quantity: 10,
    unit: 'kg',
    total_price: 800,
    status: 'pending',
    created_at: '2024-01-15T10:00:00Z',
    order_number: 'ORD-001',
    delivery_address: '456 Main Street, Dhaka',
  },
  {
    id: 2,
    buyer_name: 'City Grocery',
    product_name: 'Farm Fresh Potatoes',
    quantity: 25,
    unit: 'kg',
    total_price: 1125,
    status: 'shipped',
    created_at: '2024-01-14T14:30:00Z',
    order_number: 'ORD-002',
    delivery_address: '789 Commercial Road, Chittagong',
  },
];

export const demoFarmerProducts = [
  {
    id: 1,
    name: 'Fresh Organic Tomatoes',
    description: 'Premium organic tomatoes',
    price: 80,
    stock: 150,
    unit: 'kg',
    category: 'Vegetables',
    image: 'https://images.unsplash.com/photo-1592841200221-a6898f307baa?w=800&h=600&fit=crop',
    created_at: '2024-01-10T08:00:00Z',
  },
  {
    id: 2,
    name: 'Farm Fresh Potatoes',
    description: 'High quality potatoes',
    price: 45,
    stock: 200,
    unit: 'kg',
    category: 'Vegetables',
    image: 'https://images.unsplash.com/photo-1518977822534-7049a61ee0c2?w=800&h=600&fit=crop',
    created_at: '2024-01-08T10:00:00Z',
  },
];

export const formatCurrency = (amount) => {
  return `৳${amount.toLocaleString('en-BD')}`;
};

export const formatDate = (dateString) => {
  return new Date(dateString).toLocaleDateString('en-BD', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};

// Demo buyer dashboard data
export const demoBuyerOrders = [
  {
    id: 1,
    order_number: 'ORD-001',
    product_name: 'Fresh Organic Tomatoes',
    quantity: 10,
    unit: 'kg',
    total_price: 800,
    status: 'pending',
    created_at: '2024-01-15T10:00:00Z',
    seller_name: 'Green Valley Farm',
    delivery_address: '456 Main Street, Dhaka',
  },
  {
    id: 2,
    order_number: 'ORD-002',
    product_name: 'Farm Fresh Potatoes',
    quantity: 25,
    unit: 'kg',
    total_price: 1125,
    status: 'shipped',
    created_at: '2024-01-14T14:30:00Z',
    seller_name: 'Sunrise Agriculture',
    delivery_address: '456 Main Street, Dhaka',
  },
  {
    id: 3,
    order_number: 'ORD-003',
    product_name: 'Fresh Mangoes',
    quantity: 5,
    unit: 'kg',
    total_price: 750,
    status: 'delivered',
    created_at: '2024-01-12T09:15:00Z',
    seller_name: 'Tropical Farm',
    delivery_address: '456 Main Street, Dhaka',
  },
];


export const demoSavedAddresses = [
  {
    id: 1,
    label: 'Home',
    address: '456 Main Street, Dhaka',
    is_default: true,
  },
  {
    id: 2,
    label: 'Office',
    address: '789 Business Avenue, Gulshan-2, Dhaka',
    is_default: false,
  },
];

// Demo seller stats
export const demoSellerStats = {
  total_products: 12,
  total_orders: 45,
  total_revenue: 85000,
  pending_orders: 3,
  active_products: 10,
  average_rating: 4.7,
  total_reviews: 128,
};

// Demo seller reviews
export const demoSellerReviews = [
  {
    id: 1,
    buyer_name: 'Fresh Market Co.',
    buyer_avatar: 'https://ui-avatars.com/api/?name=Fresh+Market&background=10B981&color=FFFFFF',
    rating: 5,
    comment: 'Excellent quality tomatoes! Very fresh and delivered on time.',
    product_name: 'Fresh Organic Tomatoes',
    created_at: '2024-01-10T08:00:00Z',
  },
  {
    id: 2,
    buyer_name: 'City Grocery',
    buyer_avatar: 'https://ui-avatars.com/api/?name=City+Grocery&background=F59E0B&color=FFFFFF',
    rating: 4,
    comment: 'Good quality potatoes. Would order again.',
    product_name: 'Farm Fresh Potatoes',
    created_at: '2024-01-08T10:00:00Z',
  },
];

